evm/src/cpu/kernel/interpreter.rs

```diff
@@ -842,9 +895,10 @@ impl<'a> Interpreter<'a> {
     }
 
     fn run_mload_general(&mut self) {
-        let context = self.pop().as_usize();
-        let segment = Segment::all()[self.pop().as_usize()];
-        let offset = self.pop().as_usize();
+        let addr = self.pop();
+        let offset = addr.low_u32() as usize;
+        let segment = Segment::all()[(addr >> SEGMENT_SCALING_FACTOR).low_u32() as usize];
+        let context = (addr >> CONTEXT_SCALING_FACTOR).low_u32() as usize;
```

Comment:
This seems sufficiently common that you could consider a helper or a macro. Maybe `let (addr, offset, segment, context) = unpackify!(self.pop())`?